File: cs/Core/Model/DefaultValues.cs
namespace BinkyRailways.Core.Model
{
    public static class DefaultValues
    {
        /// <summary>
        /// Default value of <see cref="IPositionedEntity.Rotation"/>
        /// </summary>
        public const int DefaultRotation = 0;

        /// <summary>
        /// Default value of <see cref="IPositionedEntity.Locked"/>
        /// </summary>
        public const bool DefaultLocked = false;

        /// <summary>
        /// Default value of <see cref="IBlock.WaitProbability"/>
        /// </summary>
        public const int DefaultBlockWaitProbability = 100;

        /// <summary>
        /// Default value of <see cref="IBlock.MinimumWaitTime"/>
        /// </summary>
        public const int DefaultBlockMinimumWaitTime = 10;

        /// <summary>
        /// Default value of <see cref="IBlock.MaximumWaitTime"/>
        /// </summary>
        public const int DefaultBlockMaximumWaitTime = 30;

        /// <summary>
        /// Default value of <see cref="IBlock.ReverseSides"/>
        /// </summary>
        public const bool DefaultBlockReverseSides = false;

        /// <summary>
        /// Default value of <see cref="IBlock.ChangeDirection"/>
        /// </summary>
        public const ChangeDirection DefaultBlockChangeDirection = ChangeDirection.Avoid;

        /// <summary>
        /// Default value of <see cref="IBlock.ChangeDirectionReversingLocs"/>
        /// </summary>
        public const bool DefaultBlockChangeDirectionReversingLocs = false;

        /// <summary>
        /// Default value of <see cref="IBlock.StationMode"/>
        /// </summary>
        public const StationMode DefaultBlockStationMode = StationMode.Auto;

        /// <summary>
        /// Default value of <see cref="IBlockGroup.MinimumLocsInGroup"/>
        /// </summary>
        public const int DefaultBlockGroupMinimumLocsInGroup = 0;

        /// <summary>
        /// Default value of <see cref="IBlockGroup.MinimumLocsOnTrackForMinimumLocsInGroupStart"/>
        /// </summary>
        public const int DefaultBlockGroupMinimumLocsOnTrackForMinimumLocsInGroupStart = 0;

        /// <summary>
        /// Default value of <see cref="ILoc.SpeedSteps"/>
        /// </summary>
        public const int DefaultLocSpeedSteps = 128;

        /// <summary>
        /// Default value of <see cref="ILoc.SlowSpeed"/>
        /// </summary>
        public const int DefaultLocSlowSpeed = 10;

        /// <summary>
        /// Default value of <see cref="ILoc.MediumSpeed"/>
        /// </summary>
        public const int DefaultLocMediumSpeed = 50;

        /// <summary>
        /// Default value of <see cref="ILoc.MaximumSpeed"/>
        /// </summary>
        public const int DefaultLocMaximumSpeed = 100;

        /// <summary>
        /// Default value of <see cref="ILoc.ChangeDirection"/>
        /// </summary>
        public const ChangeDirection DefaultLocChangeDirection = ChangeDirection.Avoid;

        /// <summary>
        /// Default value of <see cref="ILoc.Owner"/>
        /// </summary>
        public const string DefaultLocOwner = "";

        /// <summary>
        /// Default value of <see cref="ILoc.Remarks"/>
        /// </summary>
        public const string DefaultLocRemarks = "";

        /// <summary>
        /// Default value of <see cref="IRoute.Speed"/>
        /// </summary>
        public const int DefaultRouteSpeed = 100;

        /// <summary>
        /// Default value of <see cref="IRoute.ChooseProbability"/>
        /// </summary>
        public const int DefaultRouteChooseProbability = 100;

        /// <summary>
        /// Default value of <see cref="IRoute.MaxDuration"/>
        /// </summary>
        public const int DefaultRouteMaxDuration = 60;

        /// <summary>
        /// Default value of <see cref="ITurnTable.InvertPositions"/>
        /// </summary>
        public const bool DefaultTurnTableInvertPositions = false;

        /// <summary>
        /// Default value of <see cref="ITurnTable.InvertWrite"/>
        /// </summary>
        public const bool DefaultTurnTableInvertWrite = false;

        /// <summary>
        /// Default value of <see cref="ITurnTable.InvertBusy"/>
        /// </summary>
        public const bool DefaultTurnTableInvertBusy = false;

        /// <summary>
        /// Default value of <see cref="ITurnTable.FirstPosition"/>
        /// </summary>
        public const int DefaultTurnTableFirstPosition = 1;

        /// <summary>
        /// Default value of <see cref="ITurnTable.LastPosition"/>
        /// </summary>
        public const int DefaultTurnTableLastPosition = 63;

        /// <summary>
        /// Default value of <see cref="ITurnTable.InitialPosition"/>
        /// </summary>
        public const int DefaultTurnTableInitialPosition = 1;

        /// <summary>
        /// Default value of <see cref="IBlockSignal.RedPattern"/>
        /// </summary>
        public const int DefaultBlockSignalRedPattern = 0x01;

        /// <summary>
        /// Default value of <see cref="IBlockSignal.GreenPattern"/>
        /// </summary>
        public const int DefaultBlockSignalGreenPattern = 0x02;

        /// <summary>
        /// Default value of <see cref="IBlockSignal.YellowPattern"/>
        /// </summary>
        public const int DefaultBlockSignalYellowPattern = 0x04;

        /// <summary>
        /// Default value of <see cref="IBlockSignal.WhitePattern"/>
        /// </summary>
        public const int DefaultBlockSignalWhitePattern = 0x08;

        /// <summary>
        /// Default value of <see cref="IBlockSignal.Position"/>
        /// </summary>
        public const BlockSide DefaultBlockSignalPosition = BlockSide.Front;

        /// <summary>
        /// Default value of <see cref="IBlockSignal.Type"/>
        /// </summary>
        public const BlockSignalType DefaultBlockSignalType = BlockSignalType.Entry;

        /// <summary>
        /// Default value of <see cref="IClock4StageOutput.MorningPattern"/>
        /// </summary>
        public const int DefaultClock4StageOutputMorningPattern = 0x00;

        /// <summary>
        /// Default value of <see cref="IClock4StageOutput.AfternoonPattern"/>
        /// </summary>
        public const int DefaultClock4StageOutputAfternoonPattern = 0x01;

        /// <summary>
        /// Default value of <see cref="IClock4StageOutput.EveningPattern"/>
        /// </summary>
        public const int DefaultClock4StageOutputEveningPattern = 0x02;

        /// <summary>
        /// Default value of <see cref="IClock4StageOutput.NightPattern"/>
        /// </summary>
        public const int DefaultClock4StageOutputNightPattern = 0x03;

        /// <summary>
        /// Default value of <see cref="IModuleRef.ZoomFactor"/>
        /// </summary>
        public const int DefaultModuleRefZoomFactor = 100;

        /// <summary>
        /// Default value of <see cref="IRouteEventBehavior.StateBehavior"/>
        /// </summary>
        public const RouteStateBehavior DefaultRouteEventBehaviorStateBehavior = RouteStateBehavior.NoChange;

        /// <summary>
        /// Default value of <see cref="IRouteEventBehavior.SpeedBehavior"/>
        /// </summary>
        public const LocSpeedBehavior DefaultRouteEventBehaviorSpeedBehavior = LocSpeedBehavior.Default;

        /// <summary>
        /// Default value of <see cref="IRailway.ClockSpeedFactor"/>
        /// </summary>
        public const int DefaultRailwayClockSpeedFactor = 72;

        /// <summary>
        /// Default value of <see cref="ISensor.Shape"/>
        /// </summary>
        public const Shapes DefaultSensorShape = Shapes.Circle;
    }
}

File: pkg/core/model/default_values.go
// Copyright 2020 Ewout Prangsma
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Author Ewout Prangsma
//

package model

// Default value of <see cref="IPositionedEntity.Rotation"/>
const DefaultRotation = 0

// Default value of <see cref="IPositionedEntity.Locked"/>
const DefaultLocked = false

// Default value of <see cref="IBlock.WaitProbability"/>
const DefaultBlockWaitProbability = 100

// Default value of <see cref="IBlock.MinimumWaitTime"/>
const DefaultBlockMinimumWaitTime = 10

// Default value of <see cref="IBlock.MaximumWaitTime"/>
const DefaultBlockMaximumWaitTime = 30

// Default value of <see cref="IBlock.ReverseSides"/>
const DefaultBlockReverseSides = false

// Default value of <see cref="IBlock.ChangeDirection"/>
const DefaultBlockChangeDirection = ChangeDirectionAvoid

// Default value of <see cref="IBlock.ChangeDirectionReversingLocs"/>
const DefaultBlockChangeDirectionReversingLocs = false

// Default value of <see cref="IBlock.StationMode"/>
const DefaultBlockStationMode = StationModeAuto

// Default value of <see cref="IBlockGroup.MinimumLocsInGroup"/>
const DefaultBlockGroupMinimumLocsInGroup = 0

// Default value of <see cref="IBlockGroup.MinimumLocsOnTrackForMinimumLocsInGroupStart"/>
const DefaultBlockGroupMinimumLocsOnTrackForMinimumLocsInGroupStart = 0

// Default value of <see cref="ILoc.SpeedSteps"/>
const DefaultLocSpeedSteps = 128

// Default value of <see cref="ILoc.SlowSpeed"/>
const DefaultLocSlowSpeed = 10

// Default value of <see cref="ILoc.MediumSpeed"/>
const DefaultLocMediumSpeed = 50

// Default value of <see cref="ILoc.MaximumSpeed"/>
const DefaultLocMaximumSpeed = 100

// Default value of <see cref="ILoc.ChangeDirection"/>
const DefaultLocChangeDirection = ChangeDirectionAvoid

// Default value of <see cref="ILoc.Owner"/>
const DefaultLocOwner = ""

// Default value of <see cref="ILoc.Remarks"/>
const DefaultLocRemarks = ""

// Default value of <see cref="Loc.VehicleType"/>
const DefaultLocVehicleType = VehicleTypeLoc

// Default value of <see cref="IRoute.Speed"/>
const DefaultRouteSpeed = 100

// Default value of <see cref="IRoute.ChooseProbability"/>
const DefaultRouteChooseProbability = 100

// Default value of <see cref="IRoute.MaxDuration"/>
const DefaultRouteMaxDuration = 60

// Default value of <see cref="ITurnTable.InvertPositions"/>
const DefaultTurnTableInvertPositions = false

// Default value of <see cref="ITurnTable.InvertWrite"/>
const DefaultTurnTableInvertWrite = false

// Default value of <see cref="ITurnTable.InvertBusy"/>
const DefaultTurnTableInvertBusy = false

// Default value of <see cref="ITurnTable.FirstPosition"/>
const DefaultTurnTableFirstPosition = 1

// Default value of <see cref="ITurnTable.LastPosition"/>
const DefaultTurnTableLastPosition = 63

// Default value of <see cref="ITurnTable.InitialPosition"/>
const DefaultTurnTableInitialPosition = 1

// Default value of <see cref="IBlockSignal.RedPattern"/>
const DefaultBlockSignalRedPattern = 0x01

// Default value of <see cref="IBlockSignal.GreenPattern"/>
const DefaultBlockSignalGreenPattern = 0x02

// Default value of <see cref="IBlockSignal.YellowPattern"/>
const DefaultBlockSignalYellowPattern = 0x04

// Default value of <see cref="IBlockSignal.WhitePattern"/>
const DefaultBlockSignalWhitePattern = 0x08

// Default value of <see cref="IBlockSignal.Position"/>
const DefaultBlockSignalPosition = BlockSideFront

// Default value of <see cref="IBlockSignal.Type"/>
const DefaultBlockSignalType = BlockSignalTypeEntry

// Default value of <see cref="IClock4StageOutput.MorningPattern"/>
const DefaultClock4StageOutputMorningPattern = 0x00

// Default value of <see cref="IClock4StageOutput.AfternoonPattern"/>
const DefaultClock4StageOutputAfternoonPattern = 0x01

// Default value of <see cref="IClock4StageOutput.EveningPattern"/>
const DefaultClock4StageOutputEveningPattern = 0x02

// Default value of <see cref="IClock4StageOutput.NightPattern"/>
const DefaultClock4StageOutputNightPattern = 0x03

// Default value of <see cref="IModuleRef.ZoomFactor"/>
const DefaultModuleRefZoomFactor = 100

// Default value of <see cref="IRouteEventBehavior.StateBehavior"/>
//const DefaultRouteEventBehaviorStateBehavior = RouteStateBehavior.NoChange

// Default value of <see cref="IRouteEventBehavior.SpeedBehavior"/>

//const DefaultRouteEventBehaviorSpeedBehavior = LocSpeedBehavior.Default

// Default value of <see cref="IRailway.ClockSpeedFactor"/>
const DefaultRailwayClockSpeedFactor = 72

// Default value of <see cref="ISensor.Shape"/>

//const DefaultSensorShape = Shapes.Circle
